test/cache.ts

```diff
@@ -310,3 +310,31 @@ test('does not hang on huge response', withServer, async (t, server, got) => {
 
 	t.is(body.length, bufferSize * times);
 });
+
+test('cached response ETag', withServer, async (t, server, got) => {
+	const etag = 'foobar';
+	const body = 'responseBody';
+
+	server.get('/', (request, response) => {
+		if (request.headers['if-none-match'] === etag) {
+			response.writeHead(304, {ETag: 'asdf'});
```

Comment:
I don't think you should change the etag header if you reply with 304...

Reply:
Yes, it was a copy and paste from the line of the 200 answer. I am going to remove it